elastic/kibana
# [Graph] Migrate from savedObjectsClient to dataViews and fix the displayed data view name

Closes https://github.com/elastic/kibana/issues/131378

## Summary

This PR refactors exiting code to use newer `dataViews` API and it fixes the displayed Data View name for the data source button. 

To test:
- Start ES in trial mode
- Navigate to Graph app
- Check that it's working as before and user can switch data views
